scalr/resource_scalr_agent_pool.go

```diff
@@ -103,6 +111,10 @@ func resourceScalrAgentPoolUpdate(ctx context.Context, d *schema.ResourceData, m
 
 	id := d.Id()
 
+	if d.HasChange("vcs_enabled") {
+		return diag.Errorf("Error updating agentPool %s: %v", id, "vcs_enabled attribute is readonly.")
```

Comment:
Don't do that. Mark `vcs_enabled` attribute with `ForceNew` flag instead.

Reply:
We don't have to delete the pool when changing because there might be agents there